File: 165.Compare-Version-Numbers/compare.py
"""
重要的是要补全,两个列表长度相同.

对划分后的列表0补全;
"""
class Solution:
    def compareVersion(self, version1, version2):
        """
        :type version1: str
        :type version2: str
        :rtype: int
        """
        v1_list, v2_list = version1.split("."), version2.split(".")
        v1n = [int(k) for k in v1_list]
        v2n = [int(k) for k in v2_list]
        l1, l2 = len(v1n), len(v2n)
        max_l = max(l1, l2)
        if l1 < max_l:
            v1n.extend([0] * (max_l - l1))
        if l2 < max_l:
            v2n.extend([0] * (max_l - l2))
        for i in range(max_l):
            if v1n[i] < v2n[i]:
                return -1
            elif v1n[i] > v2n[i]:
                return 1
        return 0

File: 165.Compare-Version-Numbers/compare.cpp
/**
自己分割:依据分割符;同时计算对应整数,然后比较;

1.12 1.124 -1
**/
class Solution {
public:
    int compareVersion(string version1, string version2) {
        int i=0, j=0;
        int M = version1.size(), N = version2.size();
        while(i<M || j<N){
            //根据. 分割字符串,并转换成10进制.
            int tmp1=0, tmp2=0;
            while(i<M && version1[i] != '.'){
                tmp1 = tmp1*10 + version1[i] - '0';
                i++;
            }
            while(j<N && version2[j] != '.'){
                tmp2 = tmp2*10 + version2[j] - '0';
                j++;
            }
            //判断两个数大小
            if(tmp1 > tmp2) return 1;
            else if(tmp1 < tmp2) return -1;
            //相等,更新i,j---跳过.字符
            i++;
            j++;
        }
        return 0;
    }
};
